RockSniffer/RPC/DiscordRPCHandler.cs

```diff
@@ -208,20 +227,28 @@ internal void UpdatePresence()
 
         private void Sniffer_OnSongChanged(object sender, RockSnifferLib.Events.OnSongChangedArgs e)
         {
-            songdetails = e.songDetails;
-            UpdatePresence();
+            lock (this) {
+                songdetails = e.songDetails;
+                needsUpdate = true;
+            }
         }
 
         private void Sniffer_OnMemoryReadout(object sender, RockSnifferLib.Events.OnMemoryReadoutArgs e)
         {
-            readout = e.memoryReadout;
-            UpdatePresence();
+            lock (this)
```

Comment:
I believe so, but maybe I'm missing something. If the sniffer thread changed the member while a different thread was reading from it, it would lead to strange and really hard to debug bugs. Is something like that possible?

Reply:
I suppose it could cause weirdness if the reference was swapped mid `UpdatePresence`.
If we keep locking here, add a `private readonly object` to lock instead of `this`.

Reply:
Can you explain why the `private readonly object`?

Reply:
To ensure no other class can use the same lock, it's good practice according to [lock statement guidelines](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/lock#guidelines).

Reply:
Ah, I see. I don't think it would be problem in practice, but makes sense.